packages/core/src/conflict-resolution.ts

```diff
@@ -389,4 +379,19 @@ export class ConflictResolution {
     const handler = this.handlers.get(initialState.type)
     return this.applyLogToState(handler, resetLog)
   }
+
+  /**
+   * Return `CommitData` with commit and JWS envelope, if applicable and not already present.
+   */
+  private async getCommitData(commitData: CommitData): Promise<CommitData> {
```

Comment:
the argument here should be of type `LogEntry`, right?  Since we want to be able to call this with or without the `commit` and `envelope` fields already present?

Reply:
Everywhere it's being used right now passes `CommitData` (`commit` and `envelope` are nullable) but I like the idea of passing `LogEntry`.

Reply:
Although, if the method took `CommitData` like it did originally, wouldn't it work just the same? `CommitData` extends `LogEntry` and so passing a `LogEntry` structure would just be equivalent to a `CommitData` structure without the `commit` and `envelope` fields (which are optional).

Reply:
right, that was kind of my point.  In my mind `getCommitData` converts a LogEntry (which doesn't have the two extra fields) to a CommitData (which does).  It's not a big deal either way though

Reply:
Ah, I see. It does also take `CommitData` sometimes and pass it through without changes if `commit` and `envelope` were already present (e.g. from `applyLog` after `fetchLog` has been called).

Reply:
yep, sometimes, but CommitData is LogEntry so that's fine.  In my mind CommitData is always LogEntry but LogEntry is not always CommitData.  It so happens that right now LogEntry *is* always CommitData, because the extra fields that CommitData adds are all nullable, but that's a happy accident for how things happen to work currently, not something that's necessarily true at the conceptual level regarding the relationship between the two types.  This is why I would have the arg as a LogEntry personally, but all of this is me being extremely pedantic and doesn't actually matter at all, feel free to leave it is as-is, if the structure of CommitData changes in the future in a way that breaks this, we can easily change it then.

Reply:
Yes, that sounds good. Ty for the discussion 👍🏼